ggshield/verticals/secret/secret_scan_collection.py

```diff
@@ -17,48 +19,111 @@
 from pygitguardian.models import (
     Detail,
     DiffKind,
-    Match,
     PolicyBreak,
     ScanResult,
     SecretIncident,
 )
 
 from ggshield.core.config.user_config import SecretConfig
-from ggshield.core.errors import UnexpectedError, handle_api_error
+from ggshield.core.errors import handle_api_error
 from ggshield.core.filter import is_in_ignored_matches
-from ggshield.core.lines import Line, get_lines_from_content
+from ggshield.core.lines import get_lines_from_content
 from ggshield.core.scan import Scannable
 from ggshield.utils.git_shell import Filemode
 from ggshield.verticals.secret.extended_match import ExtendedMatch
 
 
-class IgnoreReason(str, Enum):
-    IGNORED_MATCH = "ignored_match"
-    IGNORED_DETECTOR = "ignored_detector"
-    KNOWN_SECRET = "known_secret"
-    NOT_INTRODUCED = "not_introduced"
-    BACKEND_EXCLUDED = "backend_excluded"
+class IgnoreKind(str, Enum):
+    IGNORED_MATCH = "Match ignored via local .gitguardian yaml"
+    IGNORED_DETECTOR = "Detector ignored via local .gitguardian yaml"
+    KNOWN_SECRET = "Secret is known in dashboard and --ignore-known-secrets is used"
+    NOT_INTRODUCED = "Secret was not in added in commit"
+    BACKEND_EXCLUDED = "Excluded by dashboard"
+
+
+@dataclass(frozen=True)
+class IgnoreReason:
+    kind: IgnoreKind
+    detail: Optional[str] = None
+
+    def to_human_readable(self):
+        res = f"{self.kind.value}"
+        if self.detail:
+            res += f" ({self.detail})"
+        return res
 
 
 def compute_ignore_reason(
     policy_break: PolicyBreak, secret_config: SecretConfig
-) -> Optional[str]:
+) -> Optional[IgnoreReason]:
     """Computes the possible ignore reason associated with a PolicyBreak"""
     ignore_reason = None
     if policy_break.diff_kind in {DiffKind.DELETION, DiffKind.CONTEXT}:
-        ignore_reason = IgnoreReason.NOT_INTRODUCED
+        ignore_reason = IgnoreReason(IgnoreKind.NOT_INTRODUCED)
     elif policy_break.is_excluded:
-        ignore_reason = f"Excluded from backend ({policy_break.exclude_reason})"
+        ignore_reason = IgnoreReason(
+            IgnoreKind.BACKEND_EXCLUDED, policy_break.exclude_reason
+        )
     elif is_in_ignored_matches(policy_break, secret_config.ignored_matches or []):
-        ignore_reason = IgnoreReason.IGNORED_MATCH
+        ignore_reason = IgnoreReason(IgnoreKind.IGNORED_MATCH)
     elif policy_break.break_type in secret_config.ignored_detectors:
-        ignore_reason = IgnoreReason.IGNORED_DETECTOR
+        ignore_reason = IgnoreReason(IgnoreKind.IGNORED_DETECTOR)
     elif secret_config.ignore_known_secrets and policy_break.known_secret:
-        ignore_reason = IgnoreReason.KNOWN_SECRET
+        ignore_reason = IgnoreReason(IgnoreKind.KNOWN_SECRET)
 
     return ignore_reason
 
 
+@dataclass
+class Secret:
+    """GGShield specific model to handle policy-breaks.
+    Named Secret since we are dropping other kind of policy breaks.
+    """
+
+    break_type: str
```

Comment:
At some point we should rename this "detector" (maybe not today, though 😉).

Reply:
Yes I've also thought of it, even started doing so, but it was reverted at some point (probably after I got tangled up in cyclic import issues)

Reply:
Done as well :)